From 8cab4d0f15174bc810729c480759c35cca5ba94f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20M=C3=A4ki?= <sebastian@tico.fi>
Date: Wed, 3 Apr 2013 13:29:22 +0300
Subject: [PATCH] Make Sun HME cards work on host with more than 2GB of memory

---
 sunhme.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sunhme.c b/sunhme.c
index a1bff49..cdcd175 100644
--- a/sunhme.c
+++ b/sunhme.c
@@ -1263,7 +1263,7 @@ static void happy_meal_init_rings(struct happy_meal *hp)
 	for (i = 0; i < RX_RING_SIZE; i++) {
 		struct sk_buff *skb;
 
-		skb = happy_meal_alloc_skb(RX_BUF_ALLOC_SIZE, GFP_ATOMIC);
+		skb = happy_meal_alloc_skb(RX_BUF_ALLOC_SIZE,GFP_ATOMIC||__GFP_DMA);
 		if (!skb) {
 			hme_write_rxd(hp, &hb->happy_meal_rxd[i], 0, 0);
 			continue;
@@ -2022,7 +2022,7 @@ static void happy_meal_rx(struct happy_meal *hp, struct net_device *dev)
 			struct sk_buff *new_skb;
 
 			/* Now refill the entry, if we can. */
-			new_skb = happy_meal_alloc_skb(RX_BUF_ALLOC_SIZE, GFP_ATOMIC);
+			new_skb = happy_meal_alloc_skb(RX_BUF_ALLOC_SIZE,GFP_ATOMIC||__GFP_DMA);
 			if (new_skb == NULL) {
 				drops++;
 				goto drop_it;
-- 
1.8.1.4

